azure-cs-recoveryservices-backup-fileshare/MyStack.cs

```diff
@@ -0,0 +1,213 @@
+
+        //backup the fileshare using a protected item
+        var protectedItem = new AzureNative.RecoveryServices.ProtectedItem("protectedItem", new AzureNative.RecoveryServices.ProtectedItemArgs
+        {
+            ContainerName = protectionContainer.Name.Apply(t => t),
+            FabricName = "Azure",
+            Properties = new Pulumi.AzureNative.RecoveryServices.Inputs.AzureFileshareProtectedItemArgs
+            {
+                BackupManagementType = Pulumi.AzureNative.RecoveryServices.BackupManagementType.AzureStorage,
+                ContainerName = protectionContainer.Name.Apply(t => t),
+                FriendlyName = fileShare.Name.Apply(t => t),
+                PolicyId = protectionPolicy.Id.Apply(t => t),
+                SourceResourceId = storageAccount.Id.Apply(t => t),
+                WorkloadType = Pulumi.AzureNative.RecoveryServices.DataSourceType.AzureFileShare,
+                ProtectedItemType = "AzureFileShareProtectedItem",
+            },
+            /*Azure uses a list of pre-built protectable names and you must use them*/
+            ProtectedItemName = Output.Tuple(resourceGroup.Name.Apply(t => t), recoveryServicesVault.Name, fileShare.Name)
+                            .Apply(names =>
+                            {
+                                return GetProtectableItemForFileShare(names.Item1, names.Item2, names.Item3);
+                            }),
+            ResourceGroupName = resourceGroup.Name,
+            VaultName = recoveryServicesVault.Name,
+        },
+        new CustomResourceOptions { DependsOn = { recoveryServicesVault, protectionContainer, fileShare, protectionPolicy, storageAccount } });
+
+
+    }
+
+    public static async Task<string> GetProtectableItemForFileShare(string resourceGroupName, string vaultName, string fileShareName)
+    {
+        var azconfig = new Pulumi.Config("azure-native");
+        string sub = azconfig.Require("subscriptionId");
+        var clientToken = await Pulumi.AzureNative.Authorization.GetClientToken.InvokeAsync();
+
+        var tokenCreds = new Microsoft.Rest.TokenCredentials(clientToken.Token);
+        var recoveryServicesBackupClient = new RecoveryServicesBackupClient(tokenCreds);
+        recoveryServicesBackupClient.SubscriptionId = sub;
+
+        var protectableObjects = await Microsoft.Azure.Management.RecoveryServices.Backup.BackupProtectableItemsOperationsExtensions.ListAsync(recoveryServicesBackupClient.BackupProtectableItems, vaultName, resourceGroupName, "backupManagementType eq 'AzureStorage'");
+        foreach (var protectableObject in protectableObjects)
+        {
+            if (fileShareName == protectableObject.Properties.FriendlyName)
+            {
+                return protectableObject.Name;
+            }
+        }
+        return string.Empty;
+    }
+
+    /*[Output("serverName")]
+    public Output<string> ServerName { get; set; }  */
```

Comment:
Are there any relevant outputs to add to this stack?

Reply:
i've added 2 outputs